templates/ts-apollo-postgres-backend/src/index.ts

```diff
@@ -1,21 +1,51 @@
 // eslint-disable-next-line @typescript-eslint/no-require-imports
 require('dotenv').config()
-import { ApolloServer } from "apollo-server-express"
+import { ApolloServer, PubSub } from "apollo-server-express"
 import cors from "cors"
 import express from "express"
 import http from "http"
-import { printSchema } from 'graphql'
-import { createRuntime } from './runtime'
+import { buildServices as buildGraphbackServices, buildGraphback } from 'graphback'
+import { loadConfigSync } from 'graphql-config'
+import path from 'path'
+import { getConnectionDetails } from './db'
+import { PgKnexDBDataProvider } from '@graphback/runtime-knex'
+import { migrateDB, removeNonSafeOperationsFilter } from 'graphql-migrations'
 
 const app = express()
 
 app.use(cors())
 
-// connect to db
-const { schema, resolvers } = createRuntime();
+const projectConfig = loadConfigSync({
+  extensions: [
+    () => ({ name: 'graphback' })
+  ]
+}).getDefault()
+
+const graphbackConfig = projectConfig.extension('graphback');
+const model = projectConfig.loadSchemaSync(path.resolve(graphbackConfig.model));
+
+const pubSub = new PubSub();
+
+const dbConfig = getConnectionDetails()
+
+const services = buildGraphbackServices({
+  defaultDataProvider: new PgKnexDBDataProvider(dbConfig),
+  pubSub
+})
+
```

Comment:
I'll be good to be able to create runtime setups with a single line or few lines (less than this) of code. I have seen this in [OVP](https://github.com/aerogear/OpenVolunteerPlatform/pull/75#discussion_r430341699). I am wondering if the user has to care about how to load graphback config and passing it around some services to finally create `schema` and `resolvers` - what I'd really hope to see is the ability to be able to get the needed bits from graphaback i.e schema, typeDefs, resolvers, context, that could be passed directly to ApolloServer constructor. 

``` 
  const {typeDefs, schema, resolvers, graphbackContext, whetever} = graphbackApi(...)  
```

/!\PS: Ignore naming in the pseudo code above.

Reply:
Graphback config will be gone from the templates soon :)

Technically this works without it, you can simply pass your own model string directly to `buildGraphback` if you want, I just didn't want to change the template too much while this spike is still in place as it would distract from what is really changed.

> what I'd really hope to see is the ability to be able to get the needed bits from graphaback i.e schema, typeDefs, resolvers, context, that could be passed directly to ApolloServer constructor

Yep, this is what I am aiming for.

https://github.com/aerogear/graphback/pull/1324/files#diff-9d04c14f3c4e1969e7db314752020f1bR36-R39

The complicated thing about having this all in a one liner is our data providers and CRUD services all rely on having the GraphQL type in the ctor to set up table mapping information etc. 

So it is difficult for a user to provide their own custom CRUD service because to create it they need to have the GraphQLObjectType and create a new one for that model and supply it.

```ts
const myCustomNoteService = new CustomCRUDService(Note);

const { typeDefs, schema, graphbackContext } = graphbackApi(...)
```

This is why I added getter/setters to the CRUD service interface, so that the constructor only requires non-unique information to the contructor, like the dbConfig to the DataProvider, and PubSub to the CRUDService, and then the Graphback service builder would later add model information to each.

Another option is to make the service builder independent, similar to how it is now. But this would still not work since data providers require the model in the constructor.

```ts
const { typeDefs, schema } = graphbackApi(...)

const serviceBuilder = new GraphbackServiceBuilder(typeDefs, new CRUDService(...))
```

Reply:
> This is why I added getter/setters to the CRUD service interface, so that the constructor only requires non-unique information to the contructor, like the dbConfig to the DataProvider, and PubSub to the CRUDService, and then the Graphback service builder would later add model information to each.

I generally vouch for immutability as much as I can and avoid `getter/setter`.  Are they intended to be called by the end user? Or just the internal contract we'd have so that the Graphback service builder will be able to set some values to create the `schema`, `typeDefs` and resolvers? 

> Another option is to make the service builder independent, similar to how it is now. But this would still not work since data providers require the model in the constructor.

```ts
const { typeDefs, schema } = graphbackApi(...)

const serviceBuilder = new GraphbackServiceBuilder(typeDefs, new CRUDService(...))
```

From example above, will the problem be solved if we pass `CRUDServiceBuilder` instead of `new CRUDService()`?

Reply:
Yep. We need to pass specification of the Service that will be able to build service itself. This will add another abstraction

Reply:
> Yep. We need to pass specification of the Service that will be able to build service itself.

Gotcha!

> This will add another abstraction

Yeap, let's start with the setter/getter approach. 

The biggest problem I had with the runtime api is the easy of getting started - a lot of setups need to be done before and this PR is already addressing that, that's awesome.

Reply:
> Yep. We need to pass specification of the Service that will be able to build service itself. This will add another abstraction

The only problem with this that I can see is that this is an abstraction which complicates the user API. 

User wants to create a CustomNoteService, CustomCommentService (which both differ) - so they also have to create a builder class for each.

Reply:
Architecturally this is much cleaner and desired - however at a compromise of adding boilerplate to the API.